### PR TITLE
Decode raw abi encoded result using reflection

### PR DESCRIPTION
**Pull request by Stefan-Ethernal**
# Description

Simplify system_state.go, by decoding generically instead of using manual unmarshal of raw result, that is provided by `map[string]interface{}` as a return value.

# Changes include

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] Hotfix (change that solves an urgent issue, and requires immediate attention)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist

- [ ] I have assigned this PR to myself
- [ ] I have added at least 1 reviewer
- [ ] I have added the relevant labels
- [ ] I have updated the official documentation
- [ ] I have added sufficient documentation in code

## Testing

- [ ] I have tested this code with the official test suite
- [ ] I have tested this code manually

### Manual tests

Please complete this section if you ran manual tests for this functionality, otherwise delete it

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it

### CODE REVIEW

**Comment by cirko333** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by Stefan-Ethernal** [comment]
@cla_assistant recheck

**Comment by luka-ciric-ethernal** [comment]
I have read the CLA Document and I hereby sign the CLA

File: consensus/polybft/system_state/system_state_test.go [hunk 1]
```diff
@@ -174,36 +174,67 @@ func TestSystemState_GetBridgeBatchByNumber(t *testing.T) {
...
+		Receiver:           types.StringToAddress("0x518489F9ed41Fc35BCD23407C484F31897067ff0"),
 		Payload:            []byte("b3"),
 	}, sbmb.Batch.Messages[2])
 
 	require.Equal(t, [2]*big.Int{big.NewInt(300), big.NewInt(200)}, sbmb.Signature)
 	require.Equal(t, []byte("smth"), sbmb.Bitmap)
 }
 
-func TestSystemState_GetValidatorSetByNumber(t *testing.T) {
+func Test_Temporary(t *testing.T) {
```
Comment (goran-ethernal) [inline consensus/polybft/system_state/system_state_test.go]: Are we gonna remove this test, or rename it?

**Review by goran-ethernal — APPROVED** [review]

File: consensus/polybft/system_state/system_state_test.go [hunk 2]
```diff
@@ -174,36 +174,67 @@ func TestSystemState_GetBridgeBatchByNumber(t *testing.T) {
...
+				VotingPower: big.NewInt(300),
+			},
+		},
+		Signature: [2]*big.Int{big.NewInt(300), big.NewInt(200)},
+		Bitmap:    []byte("smth"),
+	}
+
+	rawSvs, err := svs.EncodeAbi()
 	require.NoError(t, err)
+	t.Log(rawSvs)
```
Comment (Stefan-Ethernal) [inline consensus/polybft/system_state/system_state_test.go]: If the test is going to remain, we do not need this log:

```suggestion
```

**Review by dusannosovic-ethernal — APPROVED** [review]

**Pull request merged**